### PR TITLE
Features/user login logout

### PR DESCRIPTION
**Pull request by garrickhenne**

### CODE REVIEW

File: frontend/src/components/NavBar.jsx [hunk 1]
```diff
@@ -1,20 +1,25 @@
...
         <NavButton name='Search' path='search' />
         <NavButton name='Reviews' path='reviews' />
-        <NavButton name='Dashboard' path='/dashboard' />
         <NavButton name='About us' path='about' />
-        <NavButton name='Sign up' path='/signup' />
-        <NavButton name='Login' path='login' />
+        {!user && <NavButton name='Sign up' path='/signup' /> }
+        {!user && <NavButton name='Login' path='login' /> }
+        {user && <NavButton name='Account' path='/dashboard' />}
+        {user && <LogoutButton />}
```
Comment (garrickhenne) [inline frontend/src/components/NavBar.jsx:22]: Change nav links based on if we are signed in

File: frontend/src/providers/AuthProvider.jsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  const userData = { user, login, logout };
+
+  return (
+    <authContext.Provider value={userData}>
+      { props.children }
+    </authContext.Provider>
+  );
+};
+
+export default AuthProvider;
```
Comment (garrickhenne) [inline frontend/src/providers/AuthProvider.jsx:36]: For security reasons from CORS or something I guess, the client is not able to access our server given cookies ie. the `req.session.user` so instead this adds a client side cookie of just the email that was returned to our login call and uses that to check if user is logged in or not.

File: frontend/src/routes/Signup.jsx [hunk 3]
```diff
@@ -0,0 +1,10 @@
+const Signup = () => {
+
+  return (
+    <div>
+      Hello world!
+    </div>
+  );
+};
+
+export default Signup;
```
Comment (garrickhenne) [inline frontend/src/routes/Signup.jsx:10]: Will be replaced with Daniels code

File: frontend/index.html [hunk 4]
```diff
@@ -1,10 +1,16 @@
...
 <html lang="en">
   <head>
     <meta charset="UTF-8" />
     <link rel="icon" type="image/svg+xml" href="/vite.svg" />
+    <link rel="preconnect" href="https://fonts.googleapis.com" />
+    <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin />
+    <link
+      href="https://fonts.googleapis.com/css2?family=Jost:ital,wght@0,100;0,200;0,300;0,400;0,500;0,600;0,700;0,800;0,900;1,100;1,200;1,300;1,400;1,500;1,600;1,700;1,800;1,900&display=swap"
+      rel="stylesheet"
+    />
```
Comment (garrickhenne) [inline frontend/index.html:11]: needed to install jost font

File: frontend/.eslintrc.cjs [hunk 5]
```diff
@@ -16,5 +16,26 @@ module.exports = {
...
+    "func-call-spacing": [
+      "error",
+      "never"
+    ],
+    "no-trailing-spaces": ["error", { "skipBlankLines": true }],
+    "func-style": ["error", "expression"],
+    "space-in-parens": "error",
+    "brace-style": "error",
+    "space-before-function-paren": ["warn", { "anonymous": "never", "named": "never", "asyncArrow": "never" }],
+    "quotes": "off",
```
Comment (garrickhenne) [inline frontend/.eslintrc.cjs:39]: added eslint rules to our frontend code similar to lhl styles

File: backend/src/index.ts [hunk 6]
```diff
@@ -1,14 +1,35 @@
...
+app.use(cookieSession({
+  name: 'session',
+  keys: [COOKIE_SESSION_KEY]
+}));
+app.use(morgan('dev'));
+const corsOptions = {
+  origin: ['http://127.0.0.1:5173', 'http://localhost:5173'],
+  credentials: true,
+  optionsSuccessStatus: 200,
+  exposedHeaders: ['set-cookie']
```
Comment (garrickhenne) [inline backend/src/index.ts:28]: Setting CORS options to play nicely

File: backend/src/index.ts [hunk 7]
```diff
@@ -1,21 +1,36 @@
...
+};
+
+app.use(cors(corsOptions));
 
-// Middleware to log incoming requests
-app.use((req, _res, next) => {
-  console.log(`Request Made: ${req.method} ${req.url}`);
-  next();
-});
```
Comment (garrickhenne) [inline backend/src/index.ts:18]: removed in place of morgan

**Review by robertshum — COMMENTED** [review]
I just realized I made comments on a closed PR whoops: https://github.com/garrickhenne/tenant/pull/24

if those issues from the closed PR are already addressed here, it's good to go!

**Comment by garrickhenne** [comment]
Removed CORS

**Review by robertshum — APPROVED** [review]

**Pull request merged**